frontend/src/components/table/ServerPaginatedDataTable.tsx

```diff
@@ -0,0 +1,198 @@
+            return (
+              <tbody>
+                {/* ts inference bug atm with useresource */}
+                {(data as unknown as Record<T, JSXChildren>[]).map((row) => (
+                  <tr>
+                    {(Object.values(row) as JSXChildren[]).map((cell) => (
+                      <td>{cell}</td>
+                    ))}
+                  </tr>
+                ))}
+              </tbody>
+            );
+          }}
+        />
+        <tfoot>
+          <div class="flex items-center gap-2">
+            {/* first page */}
+            <button
+              class="btn border rounded p-1"
+              onClick$={async () => {
+                currentPage.value = 0;
+              }}
+              disabled={isOnFirstPage}
+            >
+              {'<<'}
+            </button>
+            {/* prev page -1 */}
+            <button
+              class="btn border rounded p-1"
+              onClick$={() => {
+                setPageNumber('prev');
+              }}
+              disabled={isOnFirstPage}
+            >
+              {'<'}
+            </button>
+            {/* next page +1 */}
+            <button
+              class="btn border rounded p-1"
+              onClick$={async () => {
+                setPageNumber('next');
+              }}
+              disabled={isOnLastPage}
+            >
+              {'>'}
+            </button>
+            {/* last page */}
+            <button
+              class="btn border rounded p-1"
+              onClick$={async () => {
+                currentPage.value = maxPages.value - 1;
+              }}
+              disabled={isOnLastPage}
+            >
+              {'>>'}
+            </button>
+            {/* pages of totalpages */}
+            <span class="flex items-center gap-1">
+              <div>Page</div>
+              <strong>
+                {currentPage.value + 1} of {maxPages.value}
+              </strong>
+            </span>
+            {/* got to page input */}
+            <span class="flex items-center gap-1">
+              | Go to page:
+              <input
+                type="number"
+                value={currentPage.value + 1}
+                min="1"
+                max={maxPages.value}
+                onChange$={(e) => {
+                  const page = e.target.value ? Number(e.target.value) - 1 : 0;
+                  currentPage.value = page;
+                }}
+                class="border p-1 rounded w-16"
+              />
+            </span>
+            {/* rowsperpage select */}
+            <select
+              value={rowsPerPage.value}
+              onChange$={(ev: QwikChangeEvent<HTMLSelectElement>) => {
+                rowsPerPage.value = +ev.target.value;
+              }}
+            >
+              {[10, 20, 30, 40, 50].map((pageSize) => (
+                <option value={pageSize}>{'Show ' + pageSize}</option>
+              ))}
+            </select>
+          </div>
```

Comment:
Consider use another component / child component of table

Reply:
Its not really reusable elsewhere in the app....its tailored to the table.
should other table variants show in the future then perhaps then it should be refactored to be standalone.